include/envoy/server/transport_socket_config.h

```diff
@@ -47,23 +67,36 @@ class TransportSocketConfigFactory {
   virtual Network::TransportSocketFactoryPtr
   createTransportSocketFactory(const Protobuf::Message& config,
                                TransportSocketFactoryContext& context) PURE;
+};
 
+/**
+ * Implemented by each transport socket and registered via class RegisterFactory for downstream
+ * transport sockets.
+ */
+class DownstreamTransportSocketConfigFactory : public virtual TransportSocketConfigFactory {
+public:
   /**
-   * @return ProtobufTypes::MessagePtr create empty config proto message. The transport socket
-   *         config, which arrives in an opaque google.protobuf.Struct message, will be converted
-   *         to JSON and then parsed into this empty proto.
-   */
-  virtual ProtobufTypes::MessagePtr createEmptyConfigProto() PURE;
-
-  /**
-   * @return std::string the identifying name for a particular TransportSocketFactoryPtr
-   *         implementation produced by the factory.
+   * Create a particular downstream transport socket factory implementation.
+   * TODO(lizan): Revisit the interface when TLS sniffing and filter chain match are implemented.
+   * @param listener_name const std::string& the name of the listener.
+   * @param server_names const std::vector<std::string>& the names of the server.
```

Comment:
Mind adding a bit more detail here?
I assumed it was for SNI, then didn't know how SNI worked if you have a vector of server names.  I was going to dig through code but I think our API comments should have enough data us non-crypto folk can puzzle it out.  If it's blatantly obvious to crypto folk that may be enough =P